### ISSUE TITLE
label and maps masker should tell if no ROI is left after masking / resampling

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#6094]
We actually have something like this in one our tests:

```python
from nilearn.maskers import NiftiLabelsMasker
from nilearn.conftest import _img_labels, _shape_3d_large, _affine_mni, _rng
from nibabel import Nifti1Image

estimator = NiftiLabelsMasker(labels_img=_img_labels())

input_img = Nifti1Image(
    _rng().random(_shape_3d_large()), _affine_mni()
)

signal = estimator.fit_transform(input_img)

print(estimator.n_elements_)
print(estimator.region_names_)
print(estimator.labels_)
print(estimator.region_ids_)
print(signal)
```

this gives:

```
/home/remi-gau/github/nilearn/nilearn/tmp/empty_img.py:11: UserWarning: After resampling the label image to the data image, the following labels were removed: {1, 2, 3, 4, 5, 6, 7, 8, 9}. Label image only contains 1 labels (including background).
  signal = estimator.fit_transform(input_img)
0
{}
[0]
{'background': 0}
[]
```

We get a warning that there is nothing left but the background (this is a generic warning about what labels were dropped), but I am thinking that this should maybe be an error when it means that your data extractor won't extract ANY data.

@nilearn/core how do people feel about this one?

**Comment by bthirion** [comment]
I'd rather go with the warning.

**Comment by Remi-Gau** [comment]
> I'd rather go with the warning.

that's OK with but I think that with the nifti masker, we cannot have a case where it does not fail if it does not extract data from at least one voxel

also all maskers raise an error if an explicit mask masking all data is passed

so I am wondering whether we should strive for internal consistency

I will open a PR to at least raise a warning because I think this is a bare minimum.

**Comment by bthirion** [comment]
OK, I was not aware that the others masker would throw an error. If this is the case, it makes sense to be consistent.
